### DISCUSSION TITLE
Adding esp_netif component of ESP-IDF into ESP32 WiFi driver to enable DHCP server in Access Point mode.

### DISCUSSION THREAD

**Discussion by Qtran1511** [zephyrproject-rtos/zephyr#67585] [Q&A]
Hi, I'm currently working on a project with ESP32 chip which requires Access Point mode functionality, and I'm aware that Zephyr has not yet supported DHCP server. However, ESP-IDF does support and I'm trying to modify its WiFi driver source code in /driver/wifi/esp32/esp_wifi_drv.c to enable DHCP server, just for experiment. 

The question is how can I add "esp_netif" component from ESP-IDF? It looks like the  esp_wifi_drv.c is having esp_wifi component enabled that the compiler could see it and compile it, but when I include the "esp_netif.h", the compiler seems to see that header file but cannot actually compile the source files. I have tried to look up the CMakeLists for that WiFi driver but cannot see any clue on how Zephyr picks up ESP-IDF components. 

I really appreciate for anyone can help me out. Thank you.

**Comment by rftafas** [comment]
I think the problem on this line of work is that Zephyr has a its own network stack, while ESP-IDF uses LWIP. [PTAL the source from ESP-IDF](https://github.com/espressif/esp-idf/blob/master/components/lwip/apps/dhcpserver/dhcpserver.c).

Just a note, the driver contains a supplicant layer and all Wi-Fi stack. It is configured by Wi-Fi manager. Once the interface is set and connection established, it behaves like other interfaces. It is not the best place to put a DHCP server. Far better to do it using Zephyr net stack making sure the DHCP server app, if enabled, gets all packets incoming on port 67.

**Reply by Qtran1511** [reply]
Thanks alot for your response, I'll try working on it.
